### PR TITLE
Updating the edit form to only show upload edits before approval

### LINKED ISSUE #505: Error while editing an approved work
I came across this error when adding some tests for the edit of  an approved work.  I will be putting a pending test into a PR soon...

[pdc_describe/staging] NoMethodError: undefined method `id' for #<S3File:0x00007fe4e8789a18 @filename="10.80021/sxes-6k67/84/2022-09-05_09-31-44_834.heic", @last_modified=2022-09-29 18:28:26 UTC, @size=2746794, @checksum="59c4a9c70640148e7c8273140f5e6bac", @query_service=#<S3QueryService:0x00007fe4e8740048 @model=#<Work id: 84, work_type: "DATASET", state: "approved", collection_id: 1, created_by_user_id: 14, created_at: "2022-09-09 17:13:10.759490000 +0000", updated_at: "2022-09-29 18:28:25.950793000 +0...

## Backtrace

    line 243 of [PROJECT_ROOT]/app/models/work.rb: block in uploads_attributes
    line 241 of [PROJECT_ROOT]/app/models/work.rb: map
    line 241 of [PROJECT_ROOT]/app/models/work.rb: uploads_attributes

## Root Cause
The root cause of this issue is that we are showing uploads [here](https://github.com/pulibrary/pdc_describe/blob/main/app/views/works/_form.html.erb#L44-L49).  Not pre-curation or post-curation, but just uploads.  Post-curation files are different from pre-curation files now and can (and should not) be displayed in the same way. 

We could easily solve the bug by not displaying the files on the edit form for approved works.  What should we really do though?  I feel like displaying the files (non editable) for curators, and maybe linking to Globus if they want to modify them seems like a reasonable first step.

[View full backtrace and more info at honeybadger.io](https://app.honeybadger.io/projects/99214/faults/89273390)

### PR DESCRIPTION
**Pull request by carolyncole**
Only show the ability to edit the uploads before the Work is approved, since after it has been approved we drive everything off of S3 

Also changed test for edit of an approved work to administered by so that as an administrator I could edit my own work.

fixes #505

### CODE REVIEW

File: app/views/works/_form.html.erb [hunk 1]
```diff
@@ -44,7 +44,14 @@
       <% if !@wizard_mode %>
         <hr />
         <div class="deposit-uploads">
-          <%= render(partial: 'works/uploads_form', locals: { form: form }) %>
+          <% if @work.approved? %>
+            <section class="uploads">
+              <h2 class="h2"><%= t('works.uploads.post_curation.heading') %></h2>
+              <%= render(partial: 'works/uploads/post_curation_s3_resources') %>
+            </section>
+         <% else %>
```
Comment (mccalluc) [inline app/views/works/_form.html.erb:52]:
```suggestion
          <% else %>
```

File: app/views/works/_form.html.erb [hunk 2]
```diff
@@ -44,7 +44,14 @@
       <% if !@wizard_mode %>
         <hr />
         <div class="deposit-uploads">
-          <%= render(partial: 'works/uploads_form', locals: { form: form }) %>
+          <% if @work.approved? %>
+            <section class="uploads">
+              <h2 class="h2"><%= t('works.uploads.post_curation.heading') %></h2>
+              <%= render(partial: 'works/uploads/post_curation_s3_resources') %>
```
Comment (mccalluc) [inline app/views/works/_form.html.erb:50]: I think this is the only place this partial is used: Maybe move the h2 into it for consistency?

File: spec/system/view_data_in_s3_spec.rb [hunk 3]
```diff
@@ -83,6 +83,20 @@
           expect(page).to have_content file1.filename
           expect(page).to have_content file2.filename
         end
+
+        context "when user is a super super_admin_user" do
```
Comment (mccalluc) [inline spec/system/view_data_in_s3_spec.rb:87]:
```suggestion
        context "when user is a super_admin_user" do
```

Probably accidental repetition? Though if there are super-super users, that would be good to know.

**Review by mccalluc — APPROVED** [review]

File: app/views/works/_form.html.erb (hunk 2 above)
Comment (carolyncole) [reply app/views/works/_form.html.erb:50]: This view is utilized a couple places.  I grabbed it from the show.

**Pull request merged**